src/compressor/CompressorClient.h

```diff
@@ -0,0 +1,44 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+#include "include/buffer_fwd.h"
+#include "CompressorRegistry.h"
+#include "auth/Auth.h"
+
+class CompConnectionMeta;
+class Connection;
+
+class CompressorClient {
+    CompressorRegistry comp_registry;
+public:
+    CompressorClient(CephContext *cct) : comp_registry(cct) { 
+        comp_registry.refresh_config();
+    } 
+    virtual ~CompressorClient() {}
+
+    /// Build a compression request to begin the handshake
+    void get_comp_request(
```

Comment:
Maybe rethinking the signature of this function:
- return the pref methods, instead of using out params,
- auth_meta seems to be a pointer to a const object for this function, so can be marked as such (or just pass a "const ref to the derefed" shared ptr)
? 

And a comment that explains what are the "side effects" (eg modifying 'con_mod')